sim/core/gcd.go

```diff
@@ -15,9 +15,7 @@ func (unit *Unit) newHardcastAction(sim *Simulation) {
 	if unit.hardcastAction == nil {
 		pa := &PendingAction{
 			NextActionAt: unit.Hardcast.Expires,
-			OnAction: func(sim *Simulation) {
-				// Don't need to do anything, the Advance() call will take care of the hardcast.
-			},
+			// No action, the Advance() call will take care of the hardcast.
```

Comment:
You'd have to leave this "as is" then, which isn't much of a problem, as this case is very rarely taken. The main culprits are NewDelayedAction() and NewPeriodicAction().
Alternatively, core could add a `func DoNothing(*Simulation) {}`, but that's probably overkill.

Reply:
Reverted.